draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+JRI keywords MAY be used in data formats that are only capable of representing a subset of UTF-8 string values, with the obvious limitation that only the representable subset of JRI keyword values will be usable.
+
+As {{!RFC3987, Section 3}} defines mappings between IRI and URI syntax, JRI MAY be used in data formats that can only directly represent URIs.
+
+#### The "$" prefix
+
+JRI keywords MUST start with a "$" character.  Context specifications MAY define their own "$"-prefixed keywords, however this is NOT RECOMMENDED as future specifications that extend JRI as discussed in {{extending}} will use that prefix.[^8]
+
+[^8]: The usage of "$" as a prefix in the JSON Schema Core Vocabulary pre-dates JRI, and is expected to continue.  Some older JSON Schema proposals from immediately post-draft-04 also used a "$" prefix, but none were adopted and JSON Schema now discourages the use of that prefix.
+
+#### Exclusion of "$comment"
+
+For compatibility with JSON Schema {{?I-D.bhutton-json-schema-01, Section 8.3}}, JRI requires that a `"$comment"` keyword present in the same object as any JRI keywords MUST NOT impact the semantics of those JRI keywords.  In particular, `"$comment"` MUST NOT be used as a semantic disambiguator as described in {{disambiguator}}.
+
+However, `"$comment"` is not itself considered to be a JRI keyword.  JRI does not place any requirements on the usage, data type, syntax, or other semantics of `"$comment"` in context specifications.  JRI implementations MUST NOT require that a `"$comment"` adjacent to JRI keywords conform to JSON Schema's `"$comment"` keyword, unless explicitly required to do so by a context specification.
+
+### Evaluation order
+
+JRI behaviors, whether implemented by JRI keywords or by context specification features, MUST be evaluated in the following order when present:
+
+1. Primary resource identifiers (`"$id"`)
+1. Secondary resource identifiers (`"$anchor"`)
+1. Locations (`"$defs"`)
+1. References (`"$ref"`, `"$extRef"`)
+
+The logic of this ordering is that primary resources must be identified first as all other keywords depend having the correct base IRI.  Next, secondary resources can be identified once the primary resource is known.  Finally, nested locations which can be reference targets or have additional identifier keywords can be discovered within the primary resource.
+
+### Identification
+
+JRI offers two identification keywords:  one for primary resources, and one for secondary resources.
+
+#### `"$id"`
+
+The value of the `"$id"` property MUST be a string, and MUST be a valid IRI-reference as defined by {{!RFC3987, Section 2.2}}, and MUST NOT contain a fragment.
+
+The object containing the `"$id"` property MUST be considered to be a primary resource and to be identified by the IRI produced by resolving the IRI-reference against the current base IRI.  This IRI MUST be considered the base IRI for the newly identified primary resource, in accordance with {{!RFC3986, Section 5.1.1}}.
```

Comment:
How about "MUST be considered to be a primary resource that is identified by the IRI produced..."?

Reply:
I don't read what you put ☝️ and the original text as the same thing, but if that's what you meant, it does read better.

Reply:
What I'm going for is that the IRI produced by `$id` MUST be considered an IRI for the primary resource in question.  What I did _not_ mean to imply is that _only_ that IRI can be considered a valid identifier for it.  Does that help, or just muddy it further?

Reply:
or

> The object containing the `"$id"` property MUST be considered to be a primary resource **identifiable** by the IRI produced...

Reply:
@DavidBiesack 

> The object containing the "$id" property MUST be identified by the IRI produced by resolving the IRI-reference against the current base IRI.

This suggests to me that it can _only_ be identified by that IRI, which is not the case.  There might be valid IRIs to use that come from outside of the context of the document- it's not really within the scope of this spec but it doesn't seem like we should imply that all other possible IRIs are unusable.  Does that make sense?  Please take a look at the updated wording and see if it gives a better/more clear impression.

Reply:
(There's also language elsewhere that states a single resource may be identifiable by multiple IRIs, but a single IRI can identify at most a single resource.)